src/sqlfluff/dialects/dialect_bigquery.py

```diff
@@ -200,7 +200,10 @@
     SimpleArrayTypeGrammar=Sequence(
         "ARRAY",
         Bracketed(
-            Ref("DatatypeSegment"),
+            Sequence(
+                Ref("DatatypeSegment"),
+                Bracketed(Anything(), optional=True),  # For types like STRING(10)
```

Comment:
Using `Anything()` is very "loose". Would it be reasonable to provide something more specific to what is actually supported/expected?

Reply:
Thanks. I made it more specific in [96e1f92](https://github.com/sqlfluff/sqlfluff/pull/3735/commits/96e1f92400b53667140788d5b00d33b261e0fa6c)